File: golang/golang-programming/basics/struct.go
package basics

import "fmt"

type Person struct {
	age  int
	name string
}

func Struct() {
	fmt.Println("初始化结构体")
	p1 := Person{24, "易天"}
	p2 := Person{age: 24, name: "易天"}
	p3 := Person{age: 24}
	p4 := Person{name: "张三"}
	fmt.Println(p1, p2, p3, p4)

	p1.age = 26
	p1.name = "王五"
	fmt.Println(p1)

}

File: julia/julia-sample/type/struct.jl
struct Person
  name::String
  age::Int64
end

person = Person("jack", 18)
person.name

# pretty printing
Base.show(io::IO, p::Person) = print(io, "Person(name: $(p.name), age:$(p.age))")

# listing fieldnames
println(fieldnames(Person))

# mutable structs
mutable struct Pair
  left
  right
end

pair = Pair(0, 0)
pair.left = 1
pair.right = 2.0
pair
